### ISSUE TITLE
Add check for not valid constraints

### ISSUE THREAD

**Issue by mfvanek** [mfvanek/pg-index-health#362]
See article https://habr.com/ru/articles/800121/

```sql
SELECT
	t.relname, -- наименование отношения
    c.conname, -- наименование ограничения
    c.contype  -- тип ограничения 
FROM pg_catalog.pg_constraint AS c
INNER JOIN pg_catalog.pg_class AS t
    ON t.oid = c.conrelid AND c.contype IN ('c', 'f') 
        AND (NOT c.convalidated);
```

**Comment by BLoHny** [comment]
I have a question regarding this issue; may I ask it?

**Comment by mfvanek** [comment]
Hi @BLoHny 

> Should I add content to the final class named Validators in the validation package of the pg-index-health-model module and write test code for it?

No, you shouldn't

Please take a look at
- https://github.com/mfvanek/pg-index-health/issues/317#issuecomment-1881538672
- https://github.com/mfvanek/pg-index-health/issues/322#issue-2016404576

There I briefly described things you need to do to implement a new check

**Comment by BLoHny** [comment]
would this satisfy your request?

```sql
SELECT
    t.relname AS table_name, -- Name of the table
    c.conname AS constraint_name, -- Name of the constraint
    c.contype AS constraint_type  -- Type of the constraint
FROM pg_catalog.pg_constraint c
JOIN pg_catalog.pg_class t ON t.oid = c.conrelid
WHERE NOT c.convalidated; -- Constraints that have not yet been validated
```

**Comment by mfvanek** [comment]
> would this satisfy your request?

It's almost ready)

I think we need to keep filtering on `c.contype IN ('c', 'f')` in where-clause.
Other constraint types usually aren't interested for us.
See https://www.postgresql.org/docs/current/catalog-pg-constraint.html

Also all sql queries have to be schema-aware
https://github.com/mfvanek/pg-index-health-sql/blob/6a5b823d2f86f3fed946f073de93a20245b8d312/sql/foreign_keys_without_index.sql#L22

I use SQLFluff for analysing sql queries syntax and codestyle
https://github.com/mfvanek/pg-index-health-sql/blob/master/.github/linters/.sqlfluff
Please reformat your query with my custom rules.

P.S. Then in Java code we need to implement a new base class for database constraints.
Existing class ForeignKey should be derived from that new class
https://github.com/mfvanek/pg-index-health/blob/master/pg-index-health-model/src/main/java/io/github/mfvanek/pg/model/constraint/ForeignKey.java

**Comment by mfvanek** [comment]
> Which sql query more satisfies your request?

I like the first variant more

**Comment by BLoHny** [comment]
I have a question.
Should we now create a new class for checking whether constraints are not valid in checks.host package of pg-index-health?

**Comment by mfvanek** [comment]
@BLoHny 
>Should we now create a new class for checking whether constraints are not valid in checks.host package of pg-index-health?

Yes, we should.  
Please take a look at https://github.com/mfvanek/pg-index-health/blob/master/CONTRIBUTING.md#implementing-a-new-check

First of all you need to add a new element to https://github.com/mfvanek/pg-index-health/blob/2470e90ec25ba35468b6852c211f46cf83741d59/pg-index-health/src/main/java/io/github/mfvanek/pg/common/maintenance/Diagnostic.java#L25

**Comment by mfvanek** [comment]
Hi @BLoHny,
please take a look at updated query https://github.com/mfvanek/pg-index-health-sql/blob/master/sql/check_not_valid_constraints.sql
I've fixed it a bit: added `order by` and changed getting table name.

**Comment by mfvanek** [comment]
Hi @BLoHny,
do you plan to send PRs to starter and demo apps?
If not I'll make it myself.
I'm going to release a new version this weekend.

**Comment by BLoHny** [comment]
https://github.com/mfvanek/pg-index-health/issues/363
I was thinking about contributing to this.

**Comment by mfvanek** [comment]
@BLoHny 
I've just released version [0.11.0](https://github.com/mfvanek/pg-index-health/releases/tag/v.0.11.0)
Thank you very much for being part of it.
